src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -3071,6 +3071,18 @@ tx_gateway_frontend::get_ongoing_tx(
                 co_return tx_errc::invalid_txn_state;
             }
 
+            if (old_tx.status == tm_transaction::tx_status::ongoing) {
+                if (old_tx.tx_seq() == tx.tx_seq()) {
+                    co_return tx;
```

Comment:
please add a comment describing a situation: something like "previous leader attempted a commit/about, the operation returned indecisive error (the cached ongoing wasn't updated) but the operation actually passed (current leader observes the commit/abort)"